test/parallel/test-async-wrap-GH13045.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+const common = require('../common');
+
+// Refs: https://github.com/nodejs/node/issues/13045
+// An HTTP Agent reuses a TLSSocket, and makes a failed call to `asyncReset`.
+
+const assert = require('assert');
+const https = require('https');
+const fs = require('fs');
+
+const serverOptions = {
+  key: fs.readFileSync(`${common.fixturesDir}/keys/agent1-key.pem`),
+  cert: fs.readFileSync(`${common.fixturesDir}/keys/agent1-cert.pem`),
+  ca: fs.readFileSync(`${common.fixturesDir}/keys/ca1-cert.pem`)
+};
+
+const server = https.createServer(serverOptions, (req, res) => {
+  res.end('hello world\n');
+});
+
+server.listen(0, common.localhostIPv4, common.mustCall(function() {
+  const port = this.address().port;
+  const clientOptions = {
+    agent: new https.Agent({
+      keepAlive: true,
+      maxSockets: 50,
+      rejectUnauthorized: false,
+    }),
+    hostname: common.localhostIPv4,
+    port: port,
+    path: '/',
+    method: 'GET'
+  };
+
+  const req = https.request(clientOptions, common.mustCall((res) => {
```

Comment:
I think `method: 'GET'`` could be dropped and `.request()` changed to `.get()` to avoid the need to `req.end()`.